third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/closures/many_arguments.move

```diff
@@ -0,0 +1,34 @@
+//# publish
```

Comment:
Btw, could you also add the following transactional test case here:

```
//# publish
module 0xc0ffee::m {
    fun foo(a: u64, b: u64, c: u64, d: u64): u64 {
        a + b + c + d
    }

    fun one(): u64 {
        1
    }

    public fun test(): u64 {
        let _ = one();
        let _ = one();
        let _ = one();

        let f = |y| 0xc0ffee::m::foo(1, 1, y, y);
        f(1)
    }
}

//# run 0xc0ffee::m::test
```

This test fails with invariant violation before this fix, and runs correctly after this fix.

Reply:
Nice, will add this (CI failed, so the fix has not landed yet which is good)